File: Samples/State Pattern/Sample_1 - CS/PanicState.cs
namespace Sample
{
    internal class PanicState : State
    {
        public override void Enter()
        {
            Console.WriteLine($"[{_enemy.GetName}.PanicState]: called Enter!");
        }

        public override void Exit()
        {
            Console.WriteLine($"[{_enemy.GetName}.PanicState]: called Exit!");
        }

        public override void Update()
        {
            Console.WriteLine($"[{_enemy.GetName}.PanicState]: called Update!");
        }
    }
}

File: Samples/State Pattern/Sample_1 - CPP/PanicState.cpp
#include "PanicState.h"

void PanicState::Enter()
{
	cout << "[" << m_enemy->GetName() << ".PanicState] called Enter!" << endl;
}

void PanicState::Update()
{
	cout << "[" << m_enemy->GetName() << ".PanicState] called Update!" << endl;
}

void PanicState::Exit()
{
	cout << "[" << m_enemy->GetName() << ".PanicState] called Exit!" << endl;
}
